### PR TITLE
When adding a macro target to a package, create a dependency on swift-syntax

### PR DESCRIPTION
**Pull request by DougGregor**
When editing a package manifest to add a new macro target, also create a package dependency on swift-syntax and import CompilerPluginSupport. We need both so that the macro will build.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

File: Sources/PackageModelSyntax/AddTarget.swift [hunk 1]
```diff
@@ -225,3 +270,27 @@ fileprivate let macroTargetDependencies: [TargetDescription.Dependency] = [
...
+
+/// The package dependency for swift-syntax, for use in macros.
+fileprivate extension PackageDependency {
+    /// Source control URL for the swift-syntax package.
+    static var swiftSyntaxURL: SourceControlURL {
+        "https://github.com/apple/swift-syntax.git"
+    }
+
+    /// Package dependency on the swift-syntax package.
+    static var swiftSyntax: PackageDependency {
```
[the remark this answers is not in the document]
Comment (DougGregor) [reply Sources/PackageModelSyntax/AddTarget.swift]: Yes, I can grab it from the command state and thread it through. I've done so in https://github.com/apple/swift-package-manager/pull/7476/commits/c66eb4fd7d091a1f5f290b31b546e9ede5a5647a, thanks for the pointer!

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test Windows

**Review by bnbarham — APPROVED** [review]

**Pull request merged**